### PR TITLE
[Camel 3.7] JSON-B data format support

### PR DESCRIPTION
**Pull request by JiriOndrusek**
fixes https://github.com/apache/camel-quarkus/issues/1973

Camel dataformat for json-b doesn't contain implementation for jsonb. As you can see in `configuration.adoc` and in a test module, I've used johnzon implementation of jsonb.

@ppalaga I can imagine also creation of jsonb extension containing johnzon implementation - which would be easier to use (without reflection configuration). What do you think?

[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by jamesnetherton** [comment]
> I can imagine also creation of jsonb extension containing johnzon implementation - which would be easier to use (without reflection configuration). What do you think?

Can we not use `quarkus-jsonb`?

**Comment by JiriOndrusek** [comment]
@jamesnetherton Unfortunatelly using `quarkus-jsonb` is not a solution. `Camel-jsonb` is based on geronimo_spec (which is by the way using Johnzon as default implementation, see https://github.com/apache/geronimo-specs/blob/trunk/geronimo-jsonb_1.0_spec/src/main/java/javax/json/bind/spi/JsonbProvider.java#L30).

 `Quarkus-jsonb` is using `yasson` (see https://github.com/quarkusio/quarkus/blob/master/extensions/jsonb/runtime/pom.xml#L19). Yasson implements `javax.json.bind:javax.json.bind-api` not `org.apache.geronimo.specs:geronimo-jsonb_1.0_spec`.

**Comment by jamesnetherton** [comment]
But aren't they both implementations of JSR-367? So plugging either library in should work so long as they both conform to the spec(s).

Couldn't we exclude the `geronimo` bits and use `quarkus-jsonb`? Or am I missing something?

**Comment by JiriOndrusek** [comment]
@jamesnetherton  I agree with you. `camel-jsonb` which this extension is for is using geronimo_spec, so excluding geronimo didn't come to my mind. 

But from my POV geronimo should be removed from `camel-jsonb`, which will allow `yasson`. Then similarly as here (https://www.codeflow.site/de/article/java-json-binding-api) yasson could be used (without need of any exclusion) or johnzon (but with addition of geronimo spec)

I've tested it locally and without geronimo we can use yasson. So there is only a question, whether we want

1. exlude geronimo with yasson
2. or include geronimo with johnzon

(option 1 seems to be more usual from my POV)

**Comment by JiriOndrusek** [comment]
Do not merge this PR please, several changes will be done.

**Comment by JiriOndrusek** [comment]
Dependency `io.quarkus:quarkus-jsonb` is added to the extension, so no configuration is required from users.

PR could be merged.

File: integration-tests/jsonb/pom.xml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <!-- The following dependencies guarantee that this module is built after them. You can update them by running `mvn process-resources -Pformat -N` from the source tree root directory -->
```
Comment (jamesnetherton) [inline integration-tests/jsonb/pom.xml:92]: Just one little nit - can you remove the duplicated comment.

Comment (JiriOndrusek) [reply integration-tests/jsonb/pom.xml:77]: Thanks, I missed that - it is fixed now.

**Pull request merged**